samples/QuickstartWeatherServer/Program.cs

```diff
@@ -0,0 +1,21 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using ModelContextProtocol;
+using System.Net.Http.Headers;
+
+var builder = Host.CreateEmptyApplicationBuilder(settings: null);
+
+builder.Services.AddMcpServer()
+    .WithStdioServerTransport()
+    .WithToolsFromAssembly();
+
+builder.Services.AddSingleton(_ =>
+{
+    var client = new HttpClient() { BaseAddress = new Uri("https://api.weather.gov") };
+    client.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue("weather-tool", "1.0"));
+    return client;
+});

```

Comment:
More idiomatic might be:
```csharp
sc.ConfigureHttpClientDefaults(b => b.ConfigureHttpClient(hc =>
{
    hc.BaseAddress = new Uri("https://api.weather.gov");
    hc.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue("weather-tool", "1.0"));
}));
```
but what you have is also ok.

Reply:
No, that makes more sense. I was considering a typed http client and a factory but that is probably overkill.

Reply:
Actually, this requires us bringing in `Microsoft.Extensions.Http.Resilience`. While it's not a big dependency or anything, it's "yet another dependency" that is being added.

To keep the quickstart simple, I think it's best to leave the code as is.